dxr/plugins/core.py

```diff
@@ -306,6 +311,8 @@ def needles(self):
         extension = splitext(self.path)[1]
         if extension:
             yield 'ext', extension[1:]  # skip the period
+        if is_image(self.path):
```

Comment:
I feel like the "is it an image" test should be factored up to create a single point of truth—sometimes we test the file extension, other times the type of `file_to_index.contents` (which is currently decided by a technically incorrect but remarkable reliable heuristic: does the file contain \0?). I'm tempted to add it to FileToIndex, but that's kind of ugly—which plugin's FileToIndex would win if they disagreed? Would we just spec it to be unoverridable? Ick. Needs more thought. Do you have any good ideas?